app/controllers/employment/CompanyCarBenefitsController.scala

```diff
@@ -53,52 +58,48 @@ class CompanyCarBenefitsController @Inject()(implicit val cc: MessagesController
 
   def submit(taxYear: Int, employmentId: String): Action[AnyContent] = authAction.async { implicit user =>
     inYearAction.notInYear(taxYear) {
-      employmentSessionService.getSessionDataResult(taxYear, employmentId) {
-        case Some(data) =>
+
+      employmentSessionService.getSessionDataResult(taxYear, employmentId) { optCya =>
+        redirectBasedOnCurrentAnswers(taxYear, employmentId, optCya, EmploymentBenefitsType)(redirects(_, taxYear, employmentId)) { data =>
+
           buildForm.bindFromRequest().fold(
             formWithErrors => Future.successful(BadRequest(companyCarBenefitsView(formWithErrors, taxYear, employmentId))),
             yesNo => {
+
               val cya = data.employment
-              val updatedCyaModel: Option[EmploymentCYAModel] = {
-                cya.employmentBenefits.flatMap(_.carVanFuelModel) match {
-                  case Some(model) =>
-                    if (yesNo) {
-                      Some(cya.copy(employmentBenefits = cya.employmentBenefits.map(_.copy(
-                        carVanFuelModel = Some(model.copy(carQuestion = Some(true)))
-                      ))))
-                    } else {
-                      Some(cya.copy(employmentBenefits = cya.employmentBenefits.map(_.copy(
-                        carVanFuelModel = Some(model.copy(carQuestion = Some(false), car = None))
-                      ))))
-                    }
-                  case None =>
-                    //                  TODO: Need to potentially update this to make a cya or something
-                    None
+              val benefits = cya.employmentBenefits
+              val carVanFuel = cya.employmentBenefits.flatMap(_.carVanFuelModel)
+
+              val updatedCyaModel: EmploymentCYAModel = {
+                if (yesNo) {
+                  cya.copy(employmentBenefits = benefits.map(_.copy(carVanFuelModel = carVanFuel.map(_.copy(carQuestion = Some(true))))))
+                } else {
+                  cya.copy(employmentBenefits = benefits.map(_.copy(
+                    carVanFuelModel = carVanFuel.map(_.copy(carQuestion = Some(false), car = None, carFuelQuestion = None, carFuel = None)))))
```

Comment:
same as the question above, is it better for carFuelQuestion to be None and not false?